File: leetcode/Maximum Depth of N-ary Tree/1.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public int maxDepth(Node root) {
        if(root == null){
            return 0;
        }
        
        int max = 1;
        
        for(int i=0 ; i<root.children.size() ; i++){
            int curr = maxDepth(root.children.get(i)) + 1;
            if(curr > max){
                max = curr;
            }
        }
        
        return max;
    }
}

File: leetcode/Binary Tree Level Order Traversal/1.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun levelOrder(root: TreeNode?): List<List<Int>> {
        if (root == null) return emptyList()
        
        val list = mutableListOf(root)
        val result = mutableListOf<List<Int>>()
        while (list.isNotEmpty()) {
            val row = mutableListOf<Int>()
            for (i in 0 until list.size) {
                val curr = list[0]!!
                list.removeAt(0)

                row.add(curr.`val`)

                curr.left?.let { list.add(it) }
                curr.right?.let { list.add(it) }
            }
            result.add(row)
        }
        return result
    }
}